src/Provider/ConfigServiceProvider.php

```diff
@@ -147,13 +147,6 @@ private function loadConfig()
             ));
         }
 
-        if ($this->filename->getExtension() != 'yml') {
```

Comment:
if the file is not valid yaml, you will get an exception anyway...

in my case, having both yml and yaml would not help, since my file has no extension (it's a temporary file created by joining to yml files).

while it's a (very) good idea to put a well known extension to a file, i don't see any reason why a specific extension should be enforced.